pkg/api/validation/validation.go

```diff
@@ -2444,6 +2421,8 @@ func validatePodAffinityTerm(podAffinityTerm api.PodAffinityTerm, allowEmptyTopo
 	}
 	if len(podAffinityTerm.TopologyKey) != 0 {
 		allErrs = append(allErrs, unversionedvalidation.ValidateLabelName(podAffinityTerm.TopologyKey, fldPath.Child("topologyKey"))...)
+	} else {
```

Comment:
This `else` is not needed as it will break validation for `preferredDuringSchedulingIgnoredDuringExecution` pod anti-affinity.